mechanicalsoup/browser.py

```diff
@@ -143,9 +143,9 @@ def _build_request(self, form, url=None, **kwargs):
                 # web browsers use empty string for inputs with missing values
                 value = input.get("value", "")
 
-            if input.get("type") == "checkbox":
-                data.setdefault(name, []).append(value)
-
+            if input.get("type") == "radio":
+                # Uses the last checked radio (only one should be checked)
+                data[name] = value
```

Comment:
Shouldn't this be `[value]` instead of just `value`?

If you have one `<input type="radio">` and another input, the code is later going to attempt a `.append` on this value and will crash.

Reply:
Yes, we could do that to avoid errors similar to #158. However, if we're going to go that route, then we need to make even more changes. For example,
```html
<form action="/action_page.php">
  <input name="box" value="0"/>
  <input checked="checked" name="box" type="radio" value="1"/>
  <input type="submit" value="Submit" />
</form>
```
This should result in a `box=0&box=1` query string (setting aside the fact that this seems like a bad HTML design decision). However, with just `data[name] = [value]`, it will clobber the 0 value from the type-less input, giving only `box=1`.

If we were to change the behavior for type=radio to `data.setdefault(name, []).append(value)` like the others, then this HTML:
```html
<form action="/action_page.php">
  <input name="box" type="radio" value="0"/>
  <input checked="checked" name="box" type="radio" value="1"/>
  <input type="submit" value="Submit" />
</form>
```
would result in `box=0&box=1`, even though it should only be `box=1`.

Off the top of my head, I can't think of a way to retain the correct single-valued results for types that are only allowed to have one value, while still accounting for the possibility of poorly designed HTML like the above.

Reply:
Perhaps we need another dict `data_radio` that we can fill-in with plain `data_radio[key] = value`, and once we're done iterating through the form, copy from `data_radio` to `data` using `setdefault(..., []).append()`?

Reply:
We could do that. Are there any other elements that are forcibly unique like `radio` that we would need to account for as well?

I worry about this function (which is already not organized in the most legible way) becoming too complex.

Reply:
At some point, we'll need some refactoring (perhaps extracting this "form -> data" to a separate file/class) I guess. I never liked this function ;-).